Author: Martin Pitt <mpitt@debian.org>
Description: Put server Unix sockets into /var/run/postgresql/ by default
Forwarded: No, Debian specific configuration with postgresql-common

Using /tmp for sockets allows everyone to spoof a PostgreSQL server. Thus use
/var/run/postgresql/ for "system" clusters which run as 'postgres' (user
clusters will still use /tmp). Since system cluster are by far the common case,
set it as default.

As a consequence of this, we now need to explicitly specify the pg_regress
socket dir as /tmp, since it runs the server as the invoking user.

Index: postgresql-9.0-9.0~beta1/src/include/pg_config_manual.h
===================================================================
--- postgresql-9.0-9.0~beta1.orig/src/include/pg_config_manual.h	2010-01-07 05:53:35.000000000 +0100
+++ postgresql-9.0-9.0~beta1/src/include/pg_config_manual.h	2010-05-01 15:01:17.697230346 +0200
@@ -141,7 +141,7 @@
  * here's where to twiddle it.  You can also override this at runtime
  * with the postmaster's -k switch.
  */
-#define DEFAULT_PGSOCKET_DIR  "/tmp"
+#define DEFAULT_PGSOCKET_DIR  "/var/run/postgresql"
 
 /*
  * The random() function is expected to yield values between 0 and
Index: postgresql-9.0-9.0~beta1/src/test/regress/pg_regress.c
===================================================================
--- postgresql-9.0-9.0~beta1.orig/src/test/regress/pg_regress.c	2010-02-26 03:01:39.000000000 +0100
+++ postgresql-9.0-9.0~beta1/src/test/regress/pg_regress.c	2010-05-01 15:01:47.244728844 +0200
@@ -899,7 +899,7 @@
 
 	/* And now we can build and execute the shell command */
 	snprintf(psql_cmd, sizeof(psql_cmd),
-			 SYSTEMQUOTE "\"%s%spsql\" -X -c \"%s\" \"%s\"" SYSTEMQUOTE,
+			 SYSTEMQUOTE "\"%s%spsql\" -h /tmp -X -c \"%s\" \"%s\"" SYSTEMQUOTE,
 			 psqldir ? psqldir : "",
 			 psqldir ? "/" : "",
 			 query_escaped,
@@ -2154,7 +2154,7 @@
 		 * Check if there is a postmaster running already.
 		 */
 		snprintf(buf2, sizeof(buf2),
-				 SYSTEMQUOTE "\"%s/psql\" -X postgres <%s 2>%s" SYSTEMQUOTE,
+				 SYSTEMQUOTE "\"%s/psql\" -h /tmp -X postgres <%s 2>%s" SYSTEMQUOTE,
 				 bindir, DEVNULL, DEVNULL);
 
 		for (i = 0; i < 16; i++)
@@ -2186,7 +2186,7 @@
 		 */
 		header(_("starting postmaster"));
 		snprintf(buf, sizeof(buf),
-				 SYSTEMQUOTE "\"%s/postgres\" -D \"%s/data\" -F%s -c \"listen_addresses=%s\" > \"%s/log/postmaster.log\" 2>&1" SYSTEMQUOTE,
+				 SYSTEMQUOTE "\"%s/postgres\" -D \"%s/data\" -k /tmp -F%s -c \"listen_addresses=%s\" > \"%s/log/postmaster.log\" 2>&1" SYSTEMQUOTE,
 				 bindir, temp_install,
 				 debug ? " -d 5" : "",
 				 hostname ? hostname : "",
Index: postgresql-9.0-9.0~beta1/src/test/regress/pg_regress_main.c
===================================================================
--- postgresql-9.0-9.0~beta1.orig/src/test/regress/pg_regress_main.c	2010-01-02 17:58:16.000000000 +0100
+++ postgresql-9.0-9.0~beta1/src/test/regress/pg_regress_main.c	2010-05-01 15:01:47.244728844 +0200
@@ -59,7 +59,7 @@
 	add_stringlist_item(expectfiles, expectfile);
 
 	snprintf(psql_cmd, sizeof(psql_cmd),
-			 SYSTEMQUOTE "\"%s%spsql\" -X -a -q -d \"%s\" < \"%s\" > \"%s\" 2>&1" SYSTEMQUOTE,
+			 SYSTEMQUOTE "\"%s%spsql\" -h /tmp -X -a -q -d \"%s\" < \"%s\" > \"%s\" 2>&1" SYSTEMQUOTE,
 			 psqldir ? psqldir : "",
 			 psqldir ? "/" : "",
 			 dblist->str,
